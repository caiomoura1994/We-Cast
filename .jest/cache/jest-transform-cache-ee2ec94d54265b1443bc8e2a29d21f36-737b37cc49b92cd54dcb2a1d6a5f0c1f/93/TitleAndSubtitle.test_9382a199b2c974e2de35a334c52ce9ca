45353811094d21959fa59748a21fe8f2
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");var _react = _interopRequireDefault(require("react"));
var _TitleAndSubtitle = _interopRequireDefault(require("../TitleAndSubtitle"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));var _jsxFileName = "/Users/caiomoura/workspace/wecast/podcastmrg/src/components/small/__tests__/TitleAndSubtitle.test.js";

test('renders correctly', function () {
  var tree = _reactTestRenderer.default.create(
  _react.default.createElement(_TitleAndSubtitle.default, {
    title: 'music.title',
    subtitle: 'subtitle',
    size: "small", __source: { fileName: _jsxFileName, lineNumber: 8 } })).

  toJSON();
  expect(tree).toMatchSnapshot();
});