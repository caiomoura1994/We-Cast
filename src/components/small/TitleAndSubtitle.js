import React from 'react'
import { Text, View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types';

class TitleAndSubtitle extends React.Component {
  static defaultProps = {
    title: '',
    subtitle: '',
    size: 'small',
    styleProps: { alignItems: "center", marginTop: 4 * 3 },
    numberOfLinesTitle: 2,
    numberOfLinesSubtitle: 1
  };

  static propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    size: PropTypes.oneOf(['large', 'medium', 'small']),
    numberOfLinesTitle: PropTypes.number,
    numberOfLinesSubtitle: PropTypes.number,
    styleProps: PropTypes.object,
  }

  render() {
    const {
      title,
      subtitle,
      size,
      numberOfLinesTitle,
      numberOfLinesSubtitle,
      styleProps
    } = this.props;

    return (
      <View style={styleProps}>
        <Text
          numberOfLines={numberOfLinesTitle}
          ellipsizeMode="tail"
          style={styles[size]}
        >
          {title}
        </Text>
        <Text
          numberOfLines={numberOfLinesSubtitle}
          ellipsizeMode="tail"
          style={styles.subtitle}
          note
        >
          {subtitle}
        </Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  small: {
    // fontSize: 32,
    // fontWeight: "200",
  },
  medium: {
    // fontSize: 32,
    // fontWeight: "200",
  },
  large: {
    fontSize: 32,
    fontWeight: "200",
  },
  subtitle: {
    fontSize: 12,
    paddingTop: 4,
    color: '#e3e3e3'
  }
})

export default TitleAndSubtitle;
