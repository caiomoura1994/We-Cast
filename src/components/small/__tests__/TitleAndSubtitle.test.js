import React from 'react';
import TitleAndSubtitle from '../TitleAndSubtitle';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <TitleAndSubtitle
      title={'music.title'}
      subtitle={'subtitle'}
      size="small"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});