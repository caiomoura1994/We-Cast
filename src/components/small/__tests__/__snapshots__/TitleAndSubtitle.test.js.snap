// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 12,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={Object {}}
  >
    music.title
  </Text>
  <Text
    ellipsizeMode="tail"
    note={true}
    numberOfLines={1}
    style={
      Object {
        "color": "#e3e3e3",
        "fontSize": 12,
        "paddingTop": 4,
      }
    }
  >
    subtitle
  </Text>
</View>
`;
